types/phplint/index.d.ts

```diff
@@ -0,0 +1,32 @@
+// Type definitions for phplint 2.0
+// Project: https://github.com/wayneashleyberry/phplint
+// Definitions by: Adam Thompson-Sharpe <https://github.com/MysteryBlokHed>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 2.8
+import grunt = require('grunt');
+type IGrunt = typeof grunt;
+
+export interface Options {
+    stdout?: boolean | undefined;
+    stderr?: boolean | undefined;
+    limit?: number | undefined;
+    suppress?: boolean | undefined;
+    useCache?: boolean | undefined;
+    cacheDirName?: string | undefined;
+    tmpDir?: string | undefined;
+    phpCmd?: string | undefined;
+}
+
+export type Callback = (error: string, stdout?: string, stderr?: string) => void;
+
+export function cli(paths: string | readonly string[], options: Options, callback?: Callback): void;
+
+export function lint(files: string | readonly string[], options: Options, callback: Callback): void;
+export function lint(files: string | readonly string[], callback: Callback): void;
+
+export function clearCache(cacheDirName: string, tmpDir: string, callback: Callback): void;
+export function clearCache(callback: Callback): void;
+
+export function gruntPlugin(grunt: IGrunt): void;
+
+export {};
```

Comment:
it is UMD package. You should use pattern link this
```suggestion
declare namespace phpLint {
  interface Options {
    stdout?: boolean | undefined;
    stderr?: boolean | undefined;
    limit?: number | undefined;
    suppress?: boolean | undefined;
    useCache?: boolean | undefined;
    cacheDirName?: string | undefined;
    tmpDir?: string | undefined;
    phpCmd?: string | undefined;
  }

  type Callback = (error: string, stdout?: string, stderr?: string) => void;

  interface Plugin {
      cli(pathsArgs: string | readonly string[], options: Options, callback?: Callback): void;

      lint(files: string | readonly string[], options: Options, callback: Callback): void;
      lint(files: string | readonly string[], callback: Callback): void;

      clearCache(cacheDirName: string, tmpDir: string, callback: Callback): void;
      clearCache(callback: Callback): void;

      gruntPlugin(grunt: IGrunt): void;
  }
}

declare const phpLint: phpLint.Plugin;

export = phpLint;
```

Reply:
What's the advantage of using this syntax instead? It looks like it's doing almost the same thing as the original code, other than the Options and Callback types not being exported

Reply:
you should follow documentation and source code. Based on the one you should define type definition as UMD package. My example will export all types.
https://github.com/wayneashleyberry/phplint#node
https://github.com/wayneashleyberry/phplint/blob/main/index.js#L100
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
https://www.typescriptlang.org/docs/handbook/modules.html#code-generation-for-modules
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

Reply:
@martin-badin I thinks it's old school, cjs module, with support for `cli` interface via bin. There is no other usage support (nor content on unpkg)